models/room.go

```diff
@@ -603,3 +603,92 @@ func GetRoomsByMetric(redisClient interfaces.RedisClient, schedulerName string,
 		}
 	}
 }
+
+// GetInvalidRoomsKey gets the key for the set that keeps names of invalid rooms (rooms from old scheduler version)
+func GetInvalidRoomsKey(schedulerName string) string {
+	return fmt.Sprintf("scheduler:%s:invalidRooms", schedulerName)
+}
+
+// GetInvalidRoomsCountKey gets the key for the string that keeps count of invalid rooms
+func GetInvalidRoomsCountKey(schedulerName string) string {
+	return fmt.Sprintf("scheduler:%s:invalidRooms:count", schedulerName)
+}
+
+// SetInvalidRooms save a room in invalid redis set
+// A room is considered invalid if its version is not the scheduler current version
+func SetInvalidRooms(redisClient interfaces.RedisClient, mr *MixedMetricsReporter, schedulerName string, roomIDs []string) error {
+	pipe := redisClient.TxPipeline()
+	pipe.Del(GetInvalidRoomsKey(schedulerName))
```

Comment:
Okay, so you're making sure the key is renovated every time you set the invalid rooms?

Reply:
Yes